dvc/dependency/repo.py

```diff
@@ -1,43 +1,77 @@
 from __future__ import unicode_literals
 
-import os
 import copy
 
-from dvc.utils.compat import urlparse
-from dvc.external_repo import ExternalRepo
+from funcy import merge
+from schema import Optional
+from contextlib import contextmanager
+
+from dvc.external_repo import external_repo
+from dvc.utils.compat import str
 
 from .local import DependencyLOCAL
 
 
 class DependencyREPO(DependencyLOCAL):
     PARAM_REPO = "repo"
+    PARAM_URL = "url"
+    PARAM_REV = "rev"
+    PARAM_REV_LOCK = "rev_lock"
 
-    def __init__(self, erepo, stage, *args, **kwargs):
-        self.erepo = ExternalRepo(stage.repo.dvc_dir, **erepo)
-        super(DependencyLOCAL, self).__init__(stage, *args, **kwargs)
-
-    def _parse_path(self, remote, path):
-        self.erepo.install(self.repo.cache.local.cache_dir)
+    REPO_SCHEMA = {
+        Optional(PARAM_URL): str,
+        Optional(PARAM_REV): str,
+        Optional(PARAM_REV_LOCK): str,
+    }
 
-        out_path = os.path.join(
-            self.erepo.repo.root_dir, urlparse(path).path.lstrip("/")
-        )
+    def __init__(self, def_repo, stage, *args, **kwargs):
+        self.def_repo = def_repo
+        super(DependencyREPO, self).__init__(stage, *args, **kwargs)
 
-        out, = self.erepo.repo.find_outs_by_path(out_path)
-        self.info = copy.copy(out.info)
-        self._erepo_stage = copy.copy(out.stage.path)
-        return self.REMOTE.path_cls(out.cache_path)
+    def _parse_path(self, remote, path):
+        return None
 
     @property
     def is_in_repo(self):
         return False
 
+    def __str__(self):
+        return "{} ({})".format(self.def_path, self.def_repo[self.PARAM_URL])
+
+    @contextmanager
+    def _make_repo(self, **overrides):
+        with external_repo(**merge(self.def_repo, overrides)) as repo:
+            yield repo
+
+    def status(self):
+        with self._make_repo() as repo:
+            current = repo.find_out_by_relpath(self.def_path).info
+
+        with self._make_repo(rev_lock=None) as repo:
+            updated = repo.find_out_by_relpath(self.def_path).info
+
+        if current != updated:
+            return {str(self): "update available"}
+
+        return {}
+
+    def save(self):
+        pass
+
     def dumpd(self):
-        ret = super(DependencyLOCAL, self).dumpd()
-        ret[self.PARAM_REPO] = self.erepo.dumpd()
-        return ret
+        return {self.PARAM_PATH: self.def_path, self.PARAM_REPO: self.def_repo}
 
     def download(self, to, resume=False):
-        self.erepo.repo.fetch(self._erepo_stage)
-        to.info = copy.copy(self.info)
-        to.checkout()
+        with self._make_repo(
+            cache_dir=self.repo.cache.local.cache_dir
+        ) as repo:
+            self.def_repo[self.PARAM_REV_LOCK] = repo.scm.get_rev()
+
+            out = repo.find_out_by_relpath(self.def_path)
+            repo.fetch(out.stage.path)
+            to.info = copy.copy(out.info)
```

Comment:
Is it safe to change passed argument here?

Reply:
@Suor yes, we are effectively doing `to.info.save()`, but it will run later anyway to verify that everything is in check.